x-pack/plugins/code/public/reducers/file.ts

```diff
@@ -133,6 +137,10 @@ export const file = handleActions(
           }
         }
       }),
+    [String(dirNotFound)]: (state: FileState, action: any) =>
+      produce<FileState>(state, (draft: FileState) => {
+        draft.notFoundDirs.push(action.payload);
```

Comment:
I am not clear why do we want to add a `notFoundDirs` array here. To me for file and directory, we can just use one flag indicating if the current path is notFound or not.

Reply:
I think it makes sense that after another file request succeed, the file `isNotFound` flag would be set false. But directory request is actually a file tree request. Would it be better if we make `isNotFound` a file/directory path array instead of a simple flag?